ir/memory.cpp

```diff
@@ -254,9 +265,11 @@ expr Memory::alloc(const expr &bytes, unsigned align, bool heap) {
 
 void Memory::free(const expr &ptr) {
   Pointer p(*this, ptr);
-  state->addUB(p.get_offset() == 0);
-  state->addUB(p.is_block_alive());
-  state->addUB(p.is_at_heap());
+  auto isNullPointer = p.get_bid() == 0 && p.get_offset() == 0;
```

Comment:
maybe create a method in Pointer, since nullness checks will be frequent I guess.

Reply:
Done